test/samples/array/expected.json

```diff
@@ -3,11 +3,11 @@
 	"fields": [
 		{
 			"doc": "example array",
-			"name": "list",
+			"name": "list_required",
```

Comment:
Could we have a new test sample for the required check. I find smaller examples easier to work with when debugging problems in this codebase.